File: sdks/typescript/examples/market-data/historical_prices.ts
import pmxt from 'pmxtjs';

const main = async () => {
    const api = new pmxt.Polymarket();
    const markets = await api.getMarketsBySlug('who-will-trump-nominate-as-fed-chair');
    const warsh = markets.find(m => m.outcomes[0]?.label === 'Kevin Warsh');

    if (warsh && warsh.outcomes && warsh.outcomes[0]) {
        const history = await api.fetchOHLCV(warsh.outcomes[0].metadata.clobTokenId, {
            resolution: '1h',
            limit: 5
        });

        console.log(history);
    }
};

main();

File: sdks/python/examples/market-data/historical_prices.py
import pmxt

def main():
    api = pmxt.Polymarket()
    events = api.search_events("Who will Trump nominate as Fed Chair?")
    event = events[0]
    
    market = event.search_markets("Kevin Warsh")[0]

    # Use outcome.outcome_id for fetching historical data
    # (Polymarket: CLOB Token ID, Kalshi: Market Ticker)
    history = api.fetch_ohlcv(market.yes.outcome_id, pmxt.HistoryFilterParams(
        resolution='1h',
        limit=5
    ))
    print(history)

if __name__ == "__main__":
    main()
